### PR TITLE
Move to multi-package suite

### PR DESCRIPTION
**Pull request by zklaus**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] ~~Reset the build number to `0` (if the version changed)~~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by bouweandela** [comment]
Looks good @zklaus! Would it make sense to start a new feedstock for this? Because now it's no longer just `esmvaltool-python`, but also `esmvaltool-ncl`.

**Comment by zklaus** [comment]
> Looks good @zklaus! Would it make sense to start a new feedstock for this? Because now it's no longer just `esmvaltool-python`, but also `esmvaltool-ncl`.

Thanks. I did think about that, but kind of decided against it because once we do have that multi-output suite, this feedstock needs to be retired and starting a new feedstock means new delays because of involvement of more people, though you might argue that's a good thing.
The slightly misleading name is no practical problem since the feedstock name doesn't imply anything about the package names, though I agree that it's not ideal. But I'd rather like to try to sort the practical issues out first and then deal with the rename. If we can't rename the feedstock and need to make a new one while retiring this one, it's still the same effort.

**Comment by zklaus** [comment]
Speaking of practical things: I was just working on the Julia part and am a bit doubtful about our package here. Basically, we can install neither Julia itself nor any of it's packages via conda-forge. What we are doing instead is installing Julia by some unspecified external means and then installing a bunch of Julia packages with Julia facilities. By packaging this up as a binary conda package and telling our users to install Julia themselves, we are merely hoping that the Julia versions (including potential system library dependencies) happen to not be too different.

Does that make sense? Perhaps we should instead tell the users that if they are keen to run Julia diagnostics, they should install not only Julia themselves, but also the Julia packages by running our `install-julia-deps.sh` script.

@conda-forge/esmvaltool-python, what do you think?

PS: Long-term, we should of course improve the R and Julia packaging on conda-forge, but that is another task.

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by bouweandela** [comment]
> Perhaps we should instead tell the users that if they are keen to run Julia diagnostics, they should install not only Julia themselves, but also the Julia packages by running our install-julia-deps.sh script.

I was hoping that at some point it would be possible to install Julia from conda, but maybe you're right and we should forget about it. It's pretty niche anyway (nobody ever contributed a Julia diagnostic script after the first one). Provided the `julia` executable is available, Julia dependencies can be installed by running `esmvaltool install Julia`, see also [here](https://docs.esmvaltool.org/en/latest/quickstart/installation.html#software-installation).

The way we're installing R dependencies is similarly clunky, we only install the R interpreter from conda and then run `esmvaltool install R` to install the R dependencies from CRAN. The reason for doing is that it became impossible for conda to solve the environment if you also add all of our R dependencies from conda (and some are not available there either).

**Comment by zklaus** [comment]
> > Perhaps we should instead tell the users that if they are keen to run Julia diagnostics, they should install not only Julia themselves, but also the Julia packages by running our install-julia-deps.sh script.
> 
> I was hoping that at some point it would be possible to install Julia from conda, but maybe you're right and we should forget about it. It's pretty niche anyway (nobody ever contributed a Julia diagnostic script after the first one). Provided the `julia` executable is available, Julia dependencies can be installed by running `esmvaltool install Julia`, see also [here](https://docs.esmvaltool.org/en/latest/quickstart/installation.html#software-installation).

There is a new effort ongoing, see conda-forge/julia-feedstock#115, and with time I hope we can pick this up again. But for now, I suggest to drop it and move forward without esmvaltool-julia.

> The way we're installing R dependencies is similarly clunky, we only install the R interpreter from conda and then run `esmvaltool install R` to install the R dependencies from CRAN. The reason for doing is that it became impossible for conda to solve the environment if you also add all of our R dependencies from conda (and some are not available there either).

I am just working on this. At the moment it's only two packages that are not available, and those may be under different names, so let's see...

**Comment by zklaus** [comment]
Of the two missing R packages one, tools, is actually a base package and is included in the `r-base` package. The other, r-ncdf4.helpers, is missing indeed, but I added it as conda-forge/staged-recipes#15828. If that is added quickly, we might just be able to pull this off.

**Comment by mkitti** [comment]
Regarding the Julia feedstock, we are pretty close. We have a working build. It's more a question of if we can swap out some more vendored dependencies for conda-forge built ones.

With regard to installing the official Julia builds, I would consider use Jill.py:
https://github.com/johnnychen94/jill.py

**Comment by zklaus** [comment]
Ok, this seems actually ready :tada: @valeriupredoi, since you have been working most with conda, a review would be great. @conda-forge/esmvaltool-python, any review would be appreciated.

**Comment by zklaus** [comment]
@mkitti, thanks! I hope we can shape up the Julia part of conda-forge in the (not so distant) future. Since for us Julia is an optional dependency that is only used by few users at the moment, we already provide instructions on how to install it alongside our project, precisely using Jill :+1:. It would be nice to integrate it fully with conda-forge though.
However, in addition to Julia itself, we also need a couple of packages, namely
- `ArgParse = "c7e460c6-2fb9-53a9-8c5b-16f535851c63"-
- `Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"-
- `DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"-
- `JSON = "682c06a0-de6a-54ab-a142-c8b1cf79cde6"-
- `NetCDF = "30363a11-5582-574a-97bb-aa9a979735b9"-
- `PyCall = "438e738f-606a-5dbb-bf0a-cddfbfd45ab0"-
- `PyPlot = "d330b81b-6aea-500a-939a-2ce795aea3ee"-
- `RainFARM = "e9a4e08f-a0a3-5224-a821-6d0231c12d6b"-
- `Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"-
- `YAML = "ddb6d928-2868-570f-bddf-ab3f9cf99eb6"-

So the battle plan is to first get Julia up-to-date, then sort out the general approach to packaging Julia package (conda-forge/julia-feedstock#14), getting all required packages into conda-forge, and finally adding the julia part here to our package.

That will probably take some time :laughing:

File: recipe/meta.yaml [hunk 1]
```diff
@@ -2,88 +2,191 @@
...
+      noarch: python
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        # - {{ pin_subpackage('esmvaltool-julia', exact=True) }}
+        - {{ pin_subpackage('esmvaltool-ncl', exact=True) }}
+        - {{ pin_subpackage('esmvaltool-python', exact=True) }}
+        - {{ pin_subpackage('esmvaltool-r', exact=True) }}
```
Comment (bouweandela) [inline recipe/meta.yaml:189]: Maybe add a test for the `esmvaltool` package too, to check that it is actually possible to install all 3 subpackages into a single environment?

Comment (zklaus) [reply recipe/meta.yaml:189]: Done in the brand-new/resuscitated conda-forge/staged-recipes#14637, see comment below.

**Comment by zklaus** [comment]
After a short chat with conda-forge folks on gitter, it was decided to do this in a separate feedstock. See conda-forge/staged-recipes#14637.

**Pull request closed without merging**